### DISCUSSION TITLE
"error" is undefined?

### DISCUSSION THREAD

**Discussion by LGmatrix13** [vercel/swr#1780] [Q&A]
Hello,

I'm getting an error that `error` is undefined. Here is my fetcher:

```
export const fetcher = async (url) => {
  const response = await fetch(url);
  if (!response.ok) {
    const error = {
      info: await response.json(),
      status: response.status,
    };
    throw error;
  }
  return response.json();
};
```
Using swr:
` const { data, error } = useSWR("/api/some-endpoint", fetcher);`

Any suggestions? Thank you!

**Comment by huozhi** [comment]
Do you have a full reproduction or codesandbox link that can help us investigate? 🙏

**Reply by LGmatrix13** [reply]
I do not sorry.... 

I am, however, importing this `fetcher` function from my lib folder. Could that create an issue? Also, this `fetcher` works with reading the data and the API is not broken, the issue just lies with `error`.
